doc/source/drivers/vector/xodr.rst

```diff
@@ -0,0 +1,198 @@
+.. _vector.xodr:
+
+XODR -- OpenDRIVE Road Description Format
+=========================================
+
+.. shortname:: XODR
+
+.. build_dependencies:: libOpenDRIVE >= 0.5.0, GEOS
+
+This driver provides read access to road network elements of OpenDRIVE datasets.
+
+`OpenDRIVE <https://www.asam.net/standards/detail/opendrive/>`_ is an open industry format for lane-detailed description of road networks, commonly used in applications of the automotive and transportation systems domains. It bundles polynomial, continuous road geometry modelling with all necessary topological links and semantic information from traffic-regulating infrastructure (signs and traffic lights).
+
+Driver capabilities
+-------------------
+
+.. supports_georeferencing::
+
+Specification version
+---------------------
+
+The currently supported OpenDRIVE version is 1.4 and basically depends on what is provided by libOpenDRIVE_. 
+
+.. _libOpenDRIVE: https://github.com/pageldev/libOpenDRIVE/
+
+Supported OpenDRIVE elements
+++++++++++++++++++++++++++++
+
+The XODR driver exposes OpenDRIVE's different road elements as separate layers by converting geometric elements into 3-dimensional OGR geometry types. The following _`layer types` are currently implemented:
+
+* *ReferenceLine*: Road reference line (``<planView>``) as :cpp:class:`OGRLineString`.
+* *LaneBorder*: Outer road lane border as :cpp:class:`OGRLineString`.
+* *Lane*: Polygonal surface (TIN) of the lane mesh as :cpp:class:`OGRTriangulatedSurface`.
+* *RoadMark*: Polygonal surface (TIN) of the road mark mesh as :cpp:class:`OGRTriangulatedSurface`.
+* *RoadObject*: Polygonal surface (TIN) of the road object mesh as :cpp:class:`OGRTriangulatedSurface`.
+* *RoadSignal*: Polygonal surface (TIN) of the road signal mesh as :cpp:class:`OGRTriangulatedSurface`.
+
+Spatial reference
+-----------------
+
+By definition, OpenDRIVE geometries are always referenced in a Cartesian coordinate system which defaults to having its origin at ``(0, 0, 0)``. If real-world coordinates are used, the OpenDRIVE header provides a PROJ.4 string with the definition of a projected spatial reference system:
+
+::
+
+  <header ...>
+    <geoReference><![CDATA[+proj=tmerc +lat_0=0 +lon_0=9 +k=0.9996 +x_0=500000 +y_0=0 +datum=WGS84 +units=m +no_defs]]></geoReference>
+  </header>
+
+The XODR driver uses this PROJ definition as spatial reference for creation of all OGR geometry layers. 
+
+Limitations
+-----------
+
+By default, OpenDRIVE XML files are opened by GDAL always in UTF-8 encoding, see :cpp:func:`VSIFOpenExL`.
```

Comment:
I'm not sure to understand this sentence. Are you refering to the encoding of the filename, or the content of the file? If it is the encoding of the filename, then it is a general GDAL policy, and not appropriate documenting here.
If it is the content, then isn't it OpenDrive job to convert to UTF8 ?

Reply:
My sentence is relating to the actual content of an OpenDRIVE XML file, which not necessarily comes encoded in UTF-8. When reading through the doc of `VSIFOpenExL`/`VSIFOpenL`, I probably overread that it only tackles *filename* encoding.

Regarding the actual content encoding, libOpenDRIVE internally used pugixml which in default configuration tries to guess the input encoding (see [`load_file()`](https://github.com/DLR-TS/libOpenDRIVE/blob/9a0437f8a18d445d5c43fe2a4c9401d8a4b770f0/thirdparty/pugixml/pugixml.hpp#L1047) and [4.6. Encodings](https://pugixml.org/docs/manual.html#loading.encoding)). I will change my hint in the GDAL driver doc accordingly.

Reply:
yes, VSIFOpenExL() only cares about the encoding of the filename, which should be UTF-8.  The encoding of the content of the file is another topic, which must be dealt by upper layers.